UnitTests/GitCommandsTests/Git/Gpg/GitGpgControllerTests.cs

```diff
@@ -38,14 +31,21 @@ public void Validate_GetRevisionCommitSignatureStatusAsync_null_revision(GitComm
         [TestCase(CommitStatus.MissingPublicKey, "E")]
         [TestCase(CommitStatus.NoSignature, "N")]
         public void Validate_GetRevisionCommitSignatureStatusAsync_good_signature(CommitStatus expected, string gitCmdReturn)
+        {
+            ExecuteGetRevisionCommitSignatureStatusAsync(expected, gitCmdReturn);
+        }
+
+        private async void ExecuteGetRevisionCommitSignatureStatusAsync(CommitStatus expected, string gitCmdReturn)
         {
             var guid = Guid.NewGuid().ToString("N");
 
-            _module().RunGitCmd($"log --pretty=\"format:%G?\" -1 {guid}").Returns(gitCmdReturn);
+            GitRevision revision = new GitRevision(null, "");
 
-            var actual = _gpgController.GetRevisionCommitSignatureStatusAsync(_revision);
+            var actual = await _gpgController.GetRevisionCommitSignatureStatusAsync(revision);
+
+            _module().RunGitCmd($"log --pretty=\"format:%G?\" -1 {guid}").Returns(gitCmdReturn);
 
-            Assert.AreEqual(actual, expected);
+            Assert.AreEqual(expected, actual);
```

Comment:
I don't believe this test is correct

The line 40 generated guid, which should be passed to `Revision`'s constructor, yet it is omitted.
`GetRevisionCommitSignatureStatusAsync` executes `RunGitCmd` which expects `revision.Guid` and it would receive the empty string, resulting in incorrect result.
`_module().RunGitCmd(...).Returns(...)` is run out of sequence. This statement belongs to "arrange" group, yet it is executed in "assert" part (see [arrange-act-assert](https://msdn.microsoft.com/en-us/library/hh694602.aspx?f=255&mspperror=-2147217396#Anchor_3)).

The correct implementation looks like this:

```
        [TestCase(CommitStatus.GoodSignature, "G")]
        [TestCase(CommitStatus.SignatureError, "B")]
        [TestCase(CommitStatus.SignatureError, "U")]
        [TestCase(CommitStatus.SignatureError, "X")]
        [TestCase(CommitStatus.SignatureError, "Y")]
        [TestCase(CommitStatus.SignatureError, "R")]
        [TestCase(CommitStatus.MissingPublicKey, "E")]
        [TestCase(CommitStatus.NoSignature, "N")]
        public async Task Validate_GetRevisionCommitSignatureStatusAsync(CommitStatus expected, string gitCmdReturn)
        {
            var guid = Guid.NewGuid().ToString("N");
            var revision = new GitRevision(null, guid);
            _module().RunGitCmd($"log --pretty=\"format:%G?\" -1 {revision.Guid}").Returns(gitCmdReturn);

            var actual  = await _gpgController.GetRevisionCommitSignatureStatusAsync(revision);

            Assert.AreEqual(expected, actual);
        }
```

Reply:
fixed in commit.